crates/services/txpool_v2/src/storage/mod.rs

```diff
@@ -0,0 +1,82 @@
+use std::{
+    collections::HashSet,
+    fmt::Debug,
+    time::Instant,
+};
+
+use crate::{
+    collision_manager::CollisionReason,
+    error::Error,
+    ports::TxPoolPersistentStorage,
+};
+use fuel_core_types::services::txpool::PoolTransaction;
+
+pub mod graph;
+
+#[derive(Debug)]
+pub struct StorageData {
+    /// The transaction.
+    pub transaction: PoolTransaction,
+    /// The cumulative tip of a transaction and all of its children.
+    pub dependents_cumulative_tip: u64,
+    /// The cumulative gas of a transaction and all of its children.
+    pub dependents_cumulative_gas: u64,
+    /// Number of dependents
+    pub number_txs_in_chain: u64,
+    /// Submitted time
+    pub submitted_time: Instant,
+}
+
+pub type RemovedTransactions = Vec<PoolTransaction>;
+
+/// The storage trait is responsible for storing transactions.
+/// It has to manage somehow the dependencies between transactions.
+pub trait Storage {
```

Comment:
Do the places that depend on this trait use all of these methods?

Before you only had one `Storage` trait, but now you have multiple. Just want to make sure you didn't leave some vestigial methods on here :)

Reply:
For now they are not all used at all but the module txpool_v2 still has the : 
```
#![allow(dead_code)]
#![allow(unused)]
```
at the top of the module. And I'm in favor so letting them because they will be useful in the future followup PR. It will be catch at the specific moment we remove these "allow"s